Helm/opensearch-dashboards/templates/deployment.yaml

```diff
@@ -123,6 +123,37 @@ spec:
         envFrom:
 {{ toYaml .Values.envFrom | indent 10 }}
 {{- end }}
+
+        {{- if and .Values.probes.readiness.enabled .Values.probes.readiness.custom }}
+        readinessProbe:
+{{ .Values.probes.readiness.custom | toYaml | indent 10 }}
+        {{- else if .Values.probes.readiness.enabled }}
+        readinessProbe:
+          httpGet:
+            path: /api/status
+            port: {{ .Values.service.httpPortName | default "http" }}
+          initialDelaySeconds: 30
+          periodSeconds: 10
+        {{- end }}
+        {{- if and .Values.probes.liveness.enabled .Values.probes.liveness.custom }}
+        livenessProbe:
+{{ .Values.probes.liveness.custom | toYaml | indent 10 }}
+        {{- else if .Values.probes.liveness.enabled }}
+        livenessProbe:
+          tcpSocket:
+            port: {{ .Values.service.httpPortName | default "http" }}
+          periodSeconds: 15
+        {{- end }}
```

Comment:
I think enabling/disabling liveness probe (or any probe) shouldn't be an option here. Map the defaults from the `values.yaml` and users can then just overwrite these parts if they choose. 

I.E;

```yaml
readinessProbe:
  httpGet:
    path: /api/status
    port: http
  periodSeconds: 15
  initialDelaySeconds: 30
  failureThreshold: 3
  timeoutSecond: 1   
livenessProbe:
  tcpSocket:
     port: http
  periodSeconds: 15
  initialDelaySeconds: 30
  failureThreshold: 3
  timeoutSecond: 1 
```
We should do this for all of them. 
 
Then we just 
```
{{- with .Values.livenessProbe }}
{{- . | toYaml | nindent X }}
{{- end }}
```

Customizing can get confusing and really users should change the base params. Just maybe timeouts and things.

Reply:
@Dandy What if some users don't want to have some probe? I agree on the mandatory `readiness` though.

Reply:
@DandyDeveloper @TheAlgo 
I think being able to disable the probes is a good thing for the users to have, and by default I think not enabling the readiness is good, especially as it is dependent on their security config. If they don't have a robust way of updating their security config their readiness probe would always be failing, which is a bad experience.